File: mockmate-backend/src/main/resources/runners/java/runner_matrix.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] dims = sc.nextLine().trim().split("\\s+");
        int rows = 0;
        int cols = 0;
        try {
            rows = Integer.parseInt(dims[0]);
            cols = Integer.parseInt(dims[1]);
            if (rows < 0 || cols < 0 || rows > 10000 || cols > 10000) {
                System.err.println("Invalid matrix dimensions");
                System.exit(1);
            }
        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
            System.err.println("Failed to parse matrix dimensions");
            System.exit(1);
        }
        int[][] matrix = new int[rows][cols];
        for (int i = 0; i < rows; i++) {
            String[] row = sc.nextLine().trim()
                .split("\\s+");
            if (row.length < cols) {
                System.err.println("Incomplete row " + i);
                System.exit(1);
            }
            for (int j = 0; j < cols; j++) {
                try {
                    matrix[i][j] = Integer.parseInt(row[j]);
                } catch (NumberFormatException e) {
                    System.err.println("Invalid integer at row " + i + " col " + j + ": " + row[j]);
                    System.exit(1);
                }
            }
        }
        Solution sol = new Solution();
        Object result = sol.{{methodSignature}}(matrix);
        if (result instanceof int[][]) {
            int[][] mat = (int[][]) result;
            for (int[] row : mat) {
                StringBuilder sb = new StringBuilder();
                for (int i = 0; i < row.length; i++) {
                    if (i > 0) sb.append(" ");
                    sb.append(row[i]);
                }
                System.out.println(sb.toString().trim());
            }
        } else if (result instanceof Boolean) {
            System.out.println(
                result.toString().toLowerCase());
        } else {
            System.out.println(result);
        }
    }
}

{{USER_CODE}}

File: mockmate-backend/src/main/resources/runners/python/runner_matrix.py
import sys

def print_result(result):
    if isinstance(result, list):
        if result and isinstance(result[0], list):
            for row in result:
                print(' '.join(map(str, row)))
        else:
            print(' '.join(map(str, result)))
    elif isinstance(result, bool):
        print(str(result).lower())
    else:
        print(result)

{{USER_CODE}}

lines = sys.stdin.read().strip().split('\n')
dims = lines[0].strip().split()
rows, cols = int(dims[0]), int(dims[1])
matrix = []
for i in range(1, rows + 1):
    row = list(map(int, lines[i].strip().split()))
    matrix.append(row)
sol = Solution()
result = sol.{{methodSignature}}(matrix)
print_result(result)
